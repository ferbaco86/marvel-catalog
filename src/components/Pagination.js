import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const PaginationContainer = styled.ul`
display: flex;
width: 100%;
padding-bottom: 2rem!important;
flex-wrap: wrap;
justify-content: center;
align-items: center;
list-style-type: none;`;

const PaginationItem = styled.li`
border-width: 2px;
border-style: solid;
border-bottom-color: ${props => (props.currentPage === props.number ? '#ec1d24' : 'none')};
margin: 0.2rem;
font-family: 'Roboto Condensed', sans-serif;
`;

const PaginationButton = styled.button`
background: none;
border: none;
cursor: pointer;
&:hover {
  color: #ec1d24;
}`;

const Pagination = props => {
  const {
    charsPerPage, totalChars, paginate, currentPage,
  } = props;
  const pageNumbers = [];

  for (let index = 1; index <= Math.ceil(totalChars / charsPerPage); index += 1) {
    pageNumbers.push(index);
  }
  return (
    <PaginationContainer>
      {pageNumbers.map(number => (
        <PaginationItem
          key={number}
          currentPage={currentPage}
          number={number}
        >
          <PaginationButton type="button" onClick={() => { paginate(number); }}>
            {number}
          </PaginationButton>
        </PaginationItem>
      ))}
    </PaginationContainer>
  );
};

Pagination.propTypes = {
  charsPerPage: PropTypes.number.isRequired,
  totalChars: PropTypes.number.isRequired,
  paginate: PropTypes.func.isRequired,
  currentPage: PropTypes.number.isRequired,
};

export default Pagination;
