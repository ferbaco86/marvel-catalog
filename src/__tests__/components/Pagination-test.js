import { render, screen, act } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../../reducers/index';
import Pagination from '../../components/Pagination';

const mockPaginate = () => 'test';
const charsNum = 10;
const total = 500;
const current = 2;

describe('Rendering component', () => {
  it('creates a Pagination component', () => {
    act(() => {
      render(
        <BrowserRouter>
          <Provider store={store}>
            <Pagination
              charsPerPage={charsNum}
              totalChars={total}
              paginate={mockPaginate}
              currentPage={current}
            />
          </Provider>
        </BrowserRouter>,
      );
    });
    screen.getByText('2');
  });
});

describe('Display', () => {
  it('renders correctly', () => {
    const snap = renderer.create(
      <BrowserRouter>
        <Provider store={store}>
          <Pagination
            charsPerPage={charsNum}
            totalChars={total}
            paginate={mockPaginate}
            currentPage={current}
          />
        </Provider>
      </BrowserRouter>,
    ).toJSON();
    expect(snap).toMatchSnapshot();
  });
});
