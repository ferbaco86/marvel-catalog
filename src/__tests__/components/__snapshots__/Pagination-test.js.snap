// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display renders correctly 1`] = `
<ul
  className="sc-bdfBwQ ePfeBh"
>
  <li
    className="sc-gsTCUz dSoyXp"
  >
    <button
      className="sc-dlfnbm jYKKPb"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
  </li>
  <li
    className="sc-gsTCUz iGlfJZ"
  >
    <button
      className="sc-dlfnbm jYKKPb"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
  </li>
  <li
    className="sc-gsTCUz dSoyXp"
  >
    <button
      className="sc-dlfnbm jYKKPb"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
  </li>
  <li
    className="sc-gsTCUz dSoyXp"
  >
    <button
      className="sc-dlfnbm jYKKPb"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
  </li>
  <li
    className="sc-gsTCUz dSoyXp"
  >
    <button
      className="sc-dlfnbm jYKKPb"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
  </li>
  <li
    className="sc-gsTCUz dSoyXp"
  >
    <button
      className="sc-dlfnbm jYKKPb"
      onClick={[Function]}
      type="button"
    >
      6
    </button>
  </li>
  <li
    className="sc-gsTCUz dSoyXp"
  >
    <button
      className="sc-dlfnbm jYKKPb"
      onClick={[Function]}
      type="button"
    >
      7
    </button>
  </li>
  <li
    className="sc-gsTCUz dSoyXp"
  >
    <button
      className="sc-dlfnbm jYKKPb"
      onClick={[Function]}
      type="button"
    >
      8
    </button>
  </li>
  <li
    className="sc-gsTCUz dSoyXp"
  >
    <button
      className="sc-dlfnbm jYKKPb"
      onClick={[Function]}
      type="button"
    >
      9
    </button>
  </li>
  <li
    className="sc-gsTCUz dSoyXp"
  >
    <button
      className="sc-dlfnbm jYKKPb"
      onClick={[Function]}
      type="button"
    >
      10
    </button>
  </li>
  <li
    className="sc-gsTCUz dSoyXp"
  >
    <button
      className="sc-dlfnbm jYKKPb"
      onClick={[Function]}
      type="button"
    >
      11
    </button>
  </li>
  <li
    className="sc-gsTCUz dSoyXp"
  >
    <button
      className="sc-dlfnbm jYKKPb"
      onClick={[Function]}
      type="button"
    >
      12
    </button>
  </li>
  <li
    className="sc-gsTCUz dSoyXp"
  >
    <button
      className="sc-dlfnbm jYKKPb"
      onClick={[Function]}
      type="button"
    >
      13
    </button>
  </li>
  <li
    className="sc-gsTCUz dSoyXp"
  >
    <button
      className="sc-dlfnbm jYKKPb"
      onClick={[Function]}
      type="button"
    >
      14
    </button>
  </li>
  <li
    className="sc-gsTCUz dSoyXp"
  >
    <button
      className="sc-dlfnbm jYKKPb"
      onClick={[Function]}
      type="button"
    >
      15
    </button>
  </li>
  <li
    className="sc-gsTCUz dSoyXp"
  >
    <button
      className="sc-dlfnbm jYKKPb"
      onClick={[Function]}
      type="button"
    >
      16
    </button>
  </li>
  <li
    className="sc-gsTCUz dSoyXp"
  >
    <button
      className="sc-dlfnbm jYKKPb"
      onClick={[Function]}
      type="button"
    >
      17
    </button>
  </li>
  <li
    className="sc-gsTCUz dSoyXp"
  >
    <button
      className="sc-dlfnbm jYKKPb"
      onClick={[Function]}
      type="button"
    >
      18
    </button>
  </li>
  <li
    className="sc-gsTCUz dSoyXp"
  >
    <button
      className="sc-dlfnbm jYKKPb"
      onClick={[Function]}
      type="button"
    >
      19
    </button>
  </li>
  <li
    className="sc-gsTCUz dSoyXp"
  >
    <button
      className="sc-dlfnbm jYKKPb"
      onClick={[Function]}
      type="button"
    >
      20
    </button>
  </li>
  <li
    className="sc-gsTCUz dSoyXp"
  >
    <button
      className="sc-dlfnbm jYKKPb"
      onClick={[Function]}
      type="button"
    >
      21
    </button>
  </li>
  <li
    className="sc-gsTCUz dSoyXp"
  >
    <button
      className="sc-dlfnbm jYKKPb"
      onClick={[Function]}
      type="button"
    >
      22
    </button>
  </li>
  <li
    className="sc-gsTCUz dSoyXp"
  >
    <button
      className="sc-dlfnbm jYKKPb"
      onClick={[Function]}
      type="button"
    >
      23
    </button>
  </li>
  <li
    className="sc-gsTCUz dSoyXp"
  >
    <button
      className="sc-dlfnbm jYKKPb"
      onClick={[Function]}
      type="button"
    >
      24
    </button>
  </li>
  <li
    className="sc-gsTCUz dSoyXp"
  >
    <button
      className="sc-dlfnbm jYKKPb"
      onClick={[Function]}
      type="button"
    >
      25
    </button>
  </li>
  <li
    className="sc-gsTCUz dSoyXp"
  >
    <button
      className="sc-dlfnbm jYKKPb"
      onClick={[Function]}
      type="button"
    >
      26
    </button>
  </li>
  <li
    className="sc-gsTCUz dSoyXp"
  >
    <button
      className="sc-dlfnbm jYKKPb"
      onClick={[Function]}
      type="button"
    >
      27
    </button>
  </li>
  <li
    className="sc-gsTCUz dSoyXp"
  >
    <button
      className="sc-dlfnbm jYKKPb"
      onClick={[Function]}
      type="button"
    >
      28
    </button>
  </li>
  <li
    className="sc-gsTCUz dSoyXp"
  >
    <button
      className="sc-dlfnbm jYKKPb"
      onClick={[Function]}
      type="button"
    >
      29
    </button>
  </li>
  <li
    className="sc-gsTCUz dSoyXp"
  >
    <button
      className="sc-dlfnbm jYKKPb"
      onClick={[Function]}
      type="button"
    >
      30
    </button>
  </li>
  <li
    className="sc-gsTCUz dSoyXp"
  >
    <button
      className="sc-dlfnbm jYKKPb"
      onClick={[Function]}
      type="button"
    >
      31
    </button>
  </li>
  <li
    className="sc-gsTCUz dSoyXp"
  >
    <button
      className="sc-dlfnbm jYKKPb"
      onClick={[Function]}
      type="button"
    >
      32
    </button>
  </li>
  <li
    className="sc-gsTCUz dSoyXp"
  >
    <button
      className="sc-dlfnbm jYKKPb"
      onClick={[Function]}
      type="button"
    >
      33
    </button>
  </li>
  <li
    className="sc-gsTCUz dSoyXp"
  >
    <button
      className="sc-dlfnbm jYKKPb"
      onClick={[Function]}
      type="button"
    >
      34
    </button>
  </li>
  <li
    className="sc-gsTCUz dSoyXp"
  >
    <button
      className="sc-dlfnbm jYKKPb"
      onClick={[Function]}
      type="button"
    >
      35
    </button>
  </li>
  <li
    className="sc-gsTCUz dSoyXp"
  >
    <button
      className="sc-dlfnbm jYKKPb"
      onClick={[Function]}
      type="button"
    >
      36
    </button>
  </li>
  <li
    className="sc-gsTCUz dSoyXp"
  >
    <button
      className="sc-dlfnbm jYKKPb"
      onClick={[Function]}
      type="button"
    >
      37
    </button>
  </li>
  <li
    className="sc-gsTCUz dSoyXp"
  >
    <button
      className="sc-dlfnbm jYKKPb"
      onClick={[Function]}
      type="button"
    >
      38
    </button>
  </li>
  <li
    className="sc-gsTCUz dSoyXp"
  >
    <button
      className="sc-dlfnbm jYKKPb"
      onClick={[Function]}
      type="button"
    >
      39
    </button>
  </li>
  <li
    className="sc-gsTCUz dSoyXp"
  >
    <button
      className="sc-dlfnbm jYKKPb"
      onClick={[Function]}
      type="button"
    >
      40
    </button>
  </li>
  <li
    className="sc-gsTCUz dSoyXp"
  >
    <button
      className="sc-dlfnbm jYKKPb"
      onClick={[Function]}
      type="button"
    >
      41
    </button>
  </li>
  <li
    className="sc-gsTCUz dSoyXp"
  >
    <button
      className="sc-dlfnbm jYKKPb"
      onClick={[Function]}
      type="button"
    >
      42
    </button>
  </li>
  <li
    className="sc-gsTCUz dSoyXp"
  >
    <button
      className="sc-dlfnbm jYKKPb"
      onClick={[Function]}
      type="button"
    >
      43
    </button>
  </li>
  <li
    className="sc-gsTCUz dSoyXp"
  >
    <button
      className="sc-dlfnbm jYKKPb"
      onClick={[Function]}
      type="button"
    >
      44
    </button>
  </li>
  <li
    className="sc-gsTCUz dSoyXp"
  >
    <button
      className="sc-dlfnbm jYKKPb"
      onClick={[Function]}
      type="button"
    >
      45
    </button>
  </li>
  <li
    className="sc-gsTCUz dSoyXp"
  >
    <button
      className="sc-dlfnbm jYKKPb"
      onClick={[Function]}
      type="button"
    >
      46
    </button>
  </li>
  <li
    className="sc-gsTCUz dSoyXp"
  >
    <button
      className="sc-dlfnbm jYKKPb"
      onClick={[Function]}
      type="button"
    >
      47
    </button>
  </li>
  <li
    className="sc-gsTCUz dSoyXp"
  >
    <button
      className="sc-dlfnbm jYKKPb"
      onClick={[Function]}
      type="button"
    >
      48
    </button>
  </li>
  <li
    className="sc-gsTCUz dSoyXp"
  >
    <button
      className="sc-dlfnbm jYKKPb"
      onClick={[Function]}
      type="button"
    >
      49
    </button>
  </li>
  <li
    className="sc-gsTCUz dSoyXp"
  >
    <button
      className="sc-dlfnbm jYKKPb"
      onClick={[Function]}
      type="button"
    >
      50
    </button>
  </li>
</ul>
`;
